### PR TITLE
Add Python/C++ Solution to MIN-MEX Cut (CF)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
sorry for not doing any prs last week, caught up with other stuff

### CODE REVIEW

File: solutions/cf-1566B.mdx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	if zeros == 0:
+		print(0)
+	else:
+		l = s.find("0")
+		r = s.rfind("0")
+  
+		if r - l + 1 == zeros:
+			print(1)
+		else:
+			print(2)
```
Comment (SansPapyrus683) [inline solutions/cf-1566B.mdx:74]:
```suggestion
		print(1 if r - l + 1 == zeros else 2)
```

File: solutions/cf-1566B.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+```
+</CPPSection>
+
+<PySection>
+
+```py
+t = int(input())
+
+for i in range(t):
```
Comment (SansPapyrus683) [inline solutions/cf-1566B.mdx:61]:
```suggestion
for _ in range(int(input())):
```

File: solutions/cf-1566B.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			// first and last occurences of zero
+			int l = s.find('0');
+			int r = s.rfind('0');
+			
+			// if they're all adjacent to one another.
+			if (r - l + 1 == zeros) {
+				cout << 1 << endl;
+			} else {
+				cout << 2 << endl;
+			}
```
Comment (SansPapyrus683) [inline solutions/cf-1566B.mdx:49]:
```suggestion
			cout << r - l + 1 == zeros ? 1 : 2 << endl;
```

**Review by SansPapyrus683 — APPROVED** [review]
yeah alr

File: solutions/cf-1566B.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+title: MIN-MEX Cut
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/94803)
+
+
+# Explanation
+
+Since MEX can't be more than 2, there's only 3 possible cases. If there are no zeros, then the MEX is 0, since all substrings would be missing a 0.  If all zeros are adjacent to one another, we can cut the entire sequence out, leaving a MEX of 1. The other groups, all comprised of ones would have a MEX of 0. Leaving the answer to be $max(0, 1) = 1$. Lastly, if there are zeros and they're not adjacent to one another, then a substring like this must exist, {0, 1}, resulting in a MEX of 2.
```
Comment (SansPapyrus683) [inline solutions/cf-1566B.mdx:13]: > Since MEX can't be more than 2

explain this maybe

Comment (ryanchou-dev) [reply solutions/cf-1566B.mdx:13]: ok

**Comment by SansPapyrus683** [comment]
ryan
that wasn't what i meant
*why* can't the MEX be more than 2

**Comment by ryanchou-dev** [comment]
> ryan
> that wasn't what i meant
> _why_ can't the MEX be more than 2
> i mean i know, but other people might not

sorry about that 😔

**Comment by SansPapyrus683** [comment]
now it's good

File: solutions/cf-1566B.mdx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>
+using namespace std;
+
```
Comment (nwatx) [inline solutions/cf-1566B.mdx:26]: is there a reason why this doesn't compile when I try to submit? 

https://codeforces.com/contest/1566/submission/129216986

**Review by nwatx — CHANGES_REQUESTED** [review]

File: solutions/cf-1566B.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/cf-1566B.mdx:26]: i think it's something that sans suggested, let me fix it

**Review by HarryWangATX — APPROVED** [review]

**Review by nathangong — APPROVED** [review]

**Review by brad182 — APPROVED** [review]
c++ looks good

**Review by nwatx — APPROVED** [review]
code passes + formatting looks good

**Pull request merged**